src/reference/architecture/data-flow.rst

```diff
@@ -0,0 +1,201 @@
+Data Flow
+=========
+
+This section looks at how Cylc synchronises data between its different
+components, the transports, security and formats used to do so.
+
+.. digraph:: _
+   :align: center
+
+   Jobs -> Scheduler [label=" ZMQ\n GraphQL\n (default)"]
+
+   Scheduler -> UIS [label=" ZMQ\n GraphQL (control)\n Protobuf (data)"]
+
+   UIS -> UI [label=" Websocket\n GraphQL"]
+
+
+.. _arch protocols:
+
+Protocols, Transport & Security
+-------------------------------
+
+Cylc uses the following schemes for data and control:
+
+
+.. _arch zmq:
+
+ZMQ (Over TCP)
+^^^^^^^^^^^^^^
+
+Cylc uses `ZMQ`_ over TCP for most server side interaction.
+
+All `ZMQ`_ connections are secured by `CurveZMQ`_ which uses security keys
+which are stored on the filesystem with appropriate permissions.
+
+Where remote communication is involved Cylc synchronises the relevant keys
+to the appropriate platforms.
+
+For more information on the key files see :ref:`Authentication Files`.
+
+.. _arch ssh:
+
+SSH
+^^^
+
+SSH connections are used for some purposes such as installing the key files
+required for :ref:`arch zmq` transport.
+
+.. _arch https:
+
+HTTPS (Hypertext Transfer Protocol Secure)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+HTTPS connections are used in the browser-based `Cylc UI`_. The certificate
+used must be configured with the `Cylc UI Server`_ / `Jupyter Server`_.
+
+.. _arch wss:
+
+WSS (WebSocket Secure)
+^^^^^^^^^^^^^^^^^^^^^^
+Websocket connections are used in the browser-based `Cylc UI`_. These
+connections are upgraded from HTTPS connections and share the same security.
+
+.. _arch formats:
+
+Formats
+-------
+
+Cylc uses the following transports to communicate over these protocols.
+
+.. _arch protobuf:
+
+Protobuf
+^^^^^^^^
+
+Protobuf is both a storage utility and transport.
+
+Protobuf is used for internal data stores and the synchronisation of those data
+stores.
+
+
+.. _arch graphql:
+
+GraphQL
+^^^^^^^
+
+`GraphQL`_ is a query language which is intended to be the user-facing
+data and control interface for Cylc workflows.
+
+It contains both data and mutations (actions) defined by a self-documenting
+schema.
+
+The GraphQL servers get their data from :ref:`arch protobuf` data stores.
```

Comment:
Not an important point but... on clicking this reference we get put on the paragraph above. Wondering if it might be better ditching the link (as we are pretty much at that section anyway)? Unless the Protobuf one will get much bigger maybe?